labs/a10.md

```diff
@@ -0,0 +1,371 @@
+you don't have anything else running on that port (if you're not on campus, you can just `curl <url_of_host_machine>:5
+050`
+from your VM or another machine on campus, e.g. ssh.ocf.berkeley.edu), and see the words "**It works!**. You may need to allow
+the port 5050 on your firewall, simply run the command `ufw allow 5050`
+
+You can actually "attach" to running containers and run more commands in them, similar to how `docker run` works. Use
+the `docker exec` command:
+
+`docker exec <container_id_or_name> <command>`
+
+To stop this container, use `docker stop <container_id_or_name>`.
+
+You can restart the container using `docker restart <container_id_or_name>`.
+
+#### Questions
+1. With `httpd` running in a detached container, run `/bin/bash` on the same container and paste the output of `ps
+aux`. Observe that there's very few processes running as compared to running `ps aux` on your VM. Why is this the case?
+
+### Dungeons and `docker-compose`
+Congratulations! You've just been hired by some trash SF Bay Area tech bubble startup as their systems administrator.
+Unfortunately, both the CEO and CTO are busy handling the business side, which leaves it up to you to get their web
+application deployed using Docker and `docker-compose`.
+
+Don't worry though -- while you may not have health insurance or a nice salary, you do have some of the CTO's notes and
+equity to help you with your task. You get off BART at 12pm and enter your cramped SOMA coworking space, sitting down
+at the desk you share with the CTO while cracking open a cold LaCroix. Checking your email, you find the following
+notes from the CTO:
+
+#### About `docker-compose`
+`docker-compose` lets you define applications that require more than one container to function. For example, on a web
+application you may want your actual web application running inside of a single container, and your database running in
+a different container.
+
+Typically you define applications in terms of **services**. Again, going with the web application example, there are
+two distinct services: the app itself, and the database backing it. `docker-compose` lets you define different services
+within a [YAML](https://en.wikipedia.org/wiki/YAML) file and run the services accordingly.
+
+One of the nice things about `docker-compose` is that it automatically sets up a network for your containers in which:
+- each container for a service is on the network and reachable from other containers on the network
+- each container is discoverable on the network via its container name
+
+This means it should be pretty simple to get our web app to connect to the database.
+
+#### Installing
+`docker-compose` is really just a python package, so ideally we should just install it with `virtualenv` and `pip`.
+Feel free to do it this way if you want, but since we're a hotshot tech startup we're gonna do this the cowboy way
+(yeehaw):
```

Comment:
lol

Maybe mention that the reason you have to do this is that the Debian packaged version is too old?